files/build_templates/buffers_config.j2

```diff
@@ -21,6 +21,30 @@ def
 {%-     set switch_role      = '' %}
 {%- endif -%}
 
+{%- set PORT_ALL  = [] %}
+
+{%- if PORT is not defined %}
+    {%- if defs.generate_port_lists(PORT_ALL) %} {% endif %}
```

Comment:
we no longer have this macro. will this cause issues if device is brought up with empty ports?

Reply:
Fixed by testing whether it is defined ahead of calling it.

Reply:
Can we check if a testcase can be added for rendering template in zero port case?

Reply:
I'm afraid we can't for now.
The j2 rendering test is built on top of minigraph loading logic which assumes there always be at least one port existing in the minigraph.